handsontable/src/plugins/loading/loading.js

```diff
@@ -0,0 +1,286 @@
+ * - `title`: Loading title to display (default: 'Loading...').
+ * - `description`: Loading description to display (default: '').
+ *
+ * @example
+ *
+ * ::: only-for javascript
+ * ```js
+ * // Enable loading plugin with default options
+ * dialog: true, // Dialog plugin must be enabled
+ * loading: true,
+ *
+ * // Enable loading plugin with custom configuration
+ * dialog: true, // Dialog plugin must be enabled
+ * loading: {
+ *   icon: 'A custom loading icon in SVG format',
+ *   title: 'Custom loading title',
+ *   description: 'Custom loading description',
+ * }
+ *
+ * // Access to loading plugin instance:
+ * const loadingPlugin = hot.getPlugin('loading');
+ *
+ * // Show a loading programmatically:
+ * loadingPlugin.show();
+ *
+ * // Hide the loading programmatically:
+ * loadingPlugin.hide();
+ *
+ * // Check if dialog is visible:
+ * const isVisible = loadingPlugin.isVisible();
+ * ```
+ * :::
+ *
+ * ::: only-for react
+ * ```jsx
+ * const MyComponent = () => {
+ *   const hotRef = useRef(null);
+ *
+ *   useEffect(() => {
+ *     const hot = hotRef.current.hotInstance;
+ *     const loadingPlugin = hot.getPlugin('loading');
+ *
+ *     loadingPlugin.show();
+ *   }, []);
+ *
+ *   return (
+ *     <HotTable
+ *       ref={hotRef}
+ *       settings={{
+ *         data: data,
+ *         dialog: true, // Dialog plugin must be enabled
+ *         loading: {
+ *           icon: 'A custom loading icon in SVG format',
+ *           title: 'Custom loading title',
+ *           description: 'Custom loading description',
+ *         }
+ *       }}
+ *     />
+ *   );
+ * }
+ * ```
+ * :::
+ *
+ * ::: only-for angular
+ * ```ts
+ * hotSettings: Handsontable.GridSettings = {
+ *   data: data,
+ *   dialog: true, // Dialog plugin must be enabled
+ *   loading: {
+ *     icon: 'A custom loading icon in SVG format',
+ *     title: 'Custom loading title',
+ *     description: 'Custom loading description',
+ *   }
+ * }
+ * ```
+ *
+ * ```html
+ * <hot-table
+ *   [settings]="hotSettings">
+ * </hot-table>
+ * ```
+ * :::
+ */
+
+export class Loading extends BasePlugin {
+  static get PLUGIN_KEY() {
+    return PLUGIN_KEY;
+  }
+
+  static get PLUGIN_PRIORITY() {
+    return PLUGIN_PRIORITY;
+  }
+
+  static get DEFAULT_SETTINGS() {
+    return {
+      // eslint-disable-next-line max-len
+      icon: `<svg class="${LOADING_CLASS_NAME}__icon-svg" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 16 16"><path stroke="currentColor" stroke-width="2" d="M15 8a7 7 0 1 1-3.5-6.062"></path></svg>`,
+      title: undefined,
+      description: '',
+    };
+  }
+
+  static get SETTINGS_VALIDATORS() {
+    return {
+      icon: value => typeof value === 'string',
+      title: value => typeof value === 'string',
+      description: value => typeof value === 'string',
+    };
+  }
+
+  /**
+   * Dialog instance reference.
+   *
+   * @type {Dialog|null}
+   */
+  #dialogInstance = null;
```

Comment:
If I could suggest something, I would change the variable to `#dialogPlugin`. The `dialogInstance` can suggest that there is an instance of Dialog (internal module with the same name) and not necessarily the plugin Dialog. WDYT?